src/integrations/admin/indexing-notification-integration.php

```diff
@@ -208,6 +208,26 @@ protected function should_show_notification() {
 		return $this->indexing_integration->get_unindexed_count() > 0;
 	}
 
+	/**
+	 * Restore the notification when the indexing reason has changed.
+	 *
+	 * @param Yoast_Notification $notification The (new) indexing notification.
+	 */
+	protected function maybe_restore_notification( $notification ) {
+		$old_notification = $this->notification_center->get_notification_by_id( self::NOTIFICATION_ID );
+
+		if ( ! $old_notification ) {
+			return;
+		}
+
+		$old_notification_data = $old_notification->get_data();
+		$reason                = $this->options_helper->get( 'indexing_reason', '' );
+
+		if ( \array_key_exists( 'reason', $old_notification_data ) && $old_notification_data['reason'] !== $reason ) {
+			$this->notification_helper->restore_notification( $notification );
+		}
```

Comment:
I would move this logic to the notification center. You just want to know if the already exists and if it contains the same values. Now you make the notification integration responsible for this. 

Something like: `$this->notification_helper->has_notification( $notification );`

Reply:
This logic got superseded by new logic in the mean time, so this comment is not applicable anymore.